### PR TITLE
Automate GCloud sync tests; qase ID 39, 40, 58, 61

### PR DESCRIPTION
**Pull request by valaparthvi**
### Which issue(s) this PR fixes (optional, in fixes #<issue number>(, fixes #<issue_number>, ...) format, will close the issue(s) when PR gets merged):
Fixes #

### Checklist:
- [ ] Squashed commits into logical changes
- [ ] Documentation
- [x] GitHub Actions (if applicable) - https://github.com/rancher/hosted-providers-e2e/actions/runs/10040606737/job/27747022520

### Special notes for your reviewer:
K8s upgrade test cases are flaky and do not always pass. Once the upgrade is complete, rancher seems to revert the upgrade to the original k8s version which is why the test case fails, this is not an automation failure though.

### CODE REVIEW

File: hosted/gke/p1/sync/sync_import_test.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+var _ = Describe("SyncImport", func() {
+	for _, testData := range []struct {
+		qaseID    int64
+		isUpgrade bool
+		testBody  func(cluster *management.Cluster, client *rancher.Client)
+		testTitle string
+	}{
+		{
+			qaseID:    341,
```
Comment (valaparthvi) [inline hosted/gke/p1/sync/sync_import_test.go:23]:
```suggestion
			qaseID:    61,
```

File: hosted/gke/p1/sync/sync_import_test.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+			isUpgrade: false,
+			testBody:  syncNodepoolsCheck,
+			testTitle: "should Sync from GCE to Rancher - add/delete nodepool",
+		},
+		{
+			qaseID:    86,
```
Comment (valaparthvi) [inline hosted/gke/p1/sync/sync_import_test.go:29]:
```suggestion
			qaseID:    58,
```

File: hosted/gke/p1/sync/sync_provisioning_test.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+var _ = Describe("SyncProvisioning", func() {
+	for _, testData := range []struct {
+		qaseID    int64
+		isUpgrade bool
+		testBody  func(cluster *management.Cluster, client *rancher.Client)
+		testTitle string
+	}{
+		{
+			qaseID:    43,
```
Comment (valaparthvi) [inline hosted/gke/p1/sync/sync_provisioning_test.go:23]:
```suggestion
			qaseID:    40,
```

File: hosted/gke/p1/sync/sync_provisioning_test.go [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+			isUpgrade: false,
+			testBody:  syncNodepoolsCheck,
+			testTitle: "should Sync from GCE to Rancher - add/delete nodepool",
+		},
+		{
+			qaseID:    42,
```
Comment (valaparthvi) [inline hosted/gke/p1/sync/sync_provisioning_test.go:29]:
```suggestion
			qaseID:    39,
```

**Comment by cpinjani** [comment]
needs rebase

File: .github/workflows/gke.yaml [hunk 5]
```diff
@@ -41,7 +41,7 @@ on:
         default: 'hostname/password'
         type: string
       tests_to_run:
-        description: Tests to run (p0_provisioning/p0_importing/p1_provisioning/p1_importing/support_matrix_provisioning/support_matrix_importing/k8s_chart_support_provisioning/k8s_chart_support_importing)
+        description: Tests to run (p0_provisioning/p0_importing/p1_provisioning/p1_importing/support_matrix_provisioning/support_matrix_importing/k8s_chart_support_provisioning/k8s_chart_support_importing/sync_provisioning/sync_importing)
```
Comment (valaparthvi) [inline .github/workflows/gke.yaml:44]: Keeping these for now, will remove it once we have a generic way to run focused tests.

File: hosted/gke/helper/helper_cluster.go [hunk 6]
```diff
@@ -463,6 +465,72 @@ func ClusterExistsOnGCloud(clusterName, project, zone string) (bool, error) {
 	return false, nil
 }
 
+// AddNodePoolOnGCloud adds a nodepool to the GKE cluster via gcloud CLI
+func AddNodePoolOnGCloud(clusterName, zone, project, npName string, extraArgs ...string) error {
```
Comment (valaparthvi) [inline hosted/gke/helper/helper_cluster.go:469]: This has been modified to be the same as the one in #132.

**Review by cpinjani — APPROVED** [review]

File: .github/workflows/gke.yaml (hunk 5 above)
Comment (cpinjani) [reply .github/workflows/gke.yaml:44]: Please add an issue to track this.

**Pull request merged**